src/System.Runtime/ref/System.Runtime.cs

```diff
@@ -6279,7 +6279,7 @@ public partial class TypeDelegator : System.Reflection.TypeInfo
     public delegate bool TypeFilter(System.Type m, object filterCriteria);
     public abstract partial class TypeInfo : System.Type, System.Reflection.IReflectableType
     {
-        internal TypeInfo() { }
+        protected TypeInfo() { }
```

Comment:
Is this already protected in coreclr and uap implementation?

Reply:
Looks like it is https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/shared/System/Reflection/TypeInfo.cs#L11. 
https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/Reflection/TypeInfo.cs#L11